spec/interactor/context_spec.rb

```diff
@@ -113,41 +114,18 @@ module Interactor
         }
       end
 
-      it "updates the context" do
-        expect {
-          begin
-            context.fail!(foo: "baz")
-          rescue
-            nil
-          end
-        }.to change {
-          context.foo
-        }.from("bar").to("baz")
-      end
-
-      it "updates the context with a string key" do
-        expect {
-          begin
-            context.fail!("foo" => "baz")
-          rescue
-            nil
-          end
-        }.to change {
-          context.foo
-        }.from("bar").to("baz")
-      end
-
       it "raises failure" do
         expect {
           context.fail!
         }.to raise_error(Failure)
       end
 
-      it "makes the context available from the failure" do
+      it "makes the error_context available from the failure" do
         begin
           context.fail!
         rescue Failure => error
-          expect(error.context).to eq(context)
+          error_context.failure = @failure = true
+          expect(error.context).to eq(error_context)
```

Comment:
I don't _think_ we need the `error_context` here. I'd probably do something like
```ruby
expect(context.success?).to eq(false)
expect(context).to_not eq(error.context)
expect(error.context.foo).to eq(nil)
```
We want to make sure it does fail the original context, then we want to make sure the context and error context are not the same, and finally want to make sure the context attributes didn't get copied to the new error context.

Reply:
I added tests for the other two cases.